app/src/main/java/com/epfl/drawyourpath/database/FirebaseDatabaseUtils.kt

```diff
@@ -281,14 +281,40 @@ object FirebaseDatabaseUtils {
      * @return the tournament, or null if an error occurred
      */
     fun mapToTournament(data: DataSnapshot?): Tournament? {
+        val tournamentInfo = mapToTournamentInfo(data?.child(FirebaseKeys.TOURNAMENT_INFO))
+        val participants = getKeys(data?.child(FirebaseKeys.TOURNAMENT_PARTICIPANTS_IDS))
+        val posts = transformPostList(data?.child(FirebaseKeys.TOURNAMENT_POSTS))
+
+        if (tournamentInfo == null) {
+            Log.e(this::class.java.name, "Tournament had null values")
+            return null
+        }
+
+        return Tournament(
+            tournamentInfo.id,
+            tournamentInfo.name,
+            tournamentInfo.description,
+            tournamentInfo.creatorId,
+            tournamentInfo.startDate,
+            tournamentInfo.endDate,
+            participants,
+            posts,
+            tournamentInfo.visibility,
```

Comment:
For constructor with a lot of arguments like Tournaments I advice to write the constructor like this:
Tournaments(tournamentsId = tournamentsInfo.id, tournamentsName = tournamentInfo.name.........)
It evict some error and the code would be more easy to read.

Reply:
Excellet idea